M1FExplained/Chapter10/Exercises/Exercise05.lean

```diff
@@ -26,31 +35,150 @@ lemma part_a (m n a : ℤ) (hmn : Int.gcd m n = 1) (hm : m ∣ a) (hn : n ∣ a)
     exact h₂ 
   match hc with 
   |⟨e, he⟩ => 
+  -- e is the number such that n * e = k₁. It also is the case that (m * n) * e = a, proving our statment. 
   use e
```

Comment:
`polyrith` does it from here, but the calc proof is fine: I just thought I'd let you know,